src/pages/workspaces/workspace/Data.js

```diff
@@ -410,7 +432,8 @@ const ToolDrawer = withModalDrawer()(({
               'Workflow'
             ]),
             h(ModalToolButton, {
-              onClick: () => dataExplorerButtonEnabled && window.open(dataExplorerUrl),
+              href: dataExplorerButtonEnabled && openDataExplorerInSameTab && dataExplorerPath,
+              onClick: () => dataExplorerButtonEnabled && !openDataExplorerInSameTab && window.open(dataExplorerUrl),
```

Comment:
Instead of manually doing `window.open`, we could just conditionally add `Utils.newTabLinkProps` to the link

Reply:
Done